### DISCUSSION TITLE
Check if a certain player is online?

### DISCUSSION THREAD

**Discussion by LuanIllogical** [Pryaxis/TShock#2098] [Q&A] [answered]
Hi! I'm starting out on plugins and I wanted to know if there's a way to check if a specific player/user is online.

Thanks in advance!

**Comment by bartico6** [comment, marked as answer]
You would do so by looping over all player indexes (0 to `Terraria.Main.maxPlayers`) and pulling an object from TShock.Players for each of those indexes and seeing if it has the properties you want:

```csharp
for(int i = 0; i < Main.maxPlayers; i++)
{
    if(Main.players[i].active)
    {
        var tp = TShock.Players[i];
        //do check logic here
    }
}
```

**Reply by bartico6** [reply]
*note this is typed up on the spot and might not be fully correct, but the general rule of thumb is to check TShock.Players, either by using a forloop over player indexes or using some sort of LINQ extension like .Where() or .Single()*

**Reply by LuanIllogical** [reply]
Thanks! The "check logic" would be checking the player's name or what? (Sorry if this is an obvious question I'm just starting out.)

**Reply by bartico6** [reply]
Well, that's up to you - "check logic" is what you want to use to check if the player is the one you're looking for: So, maybe a name, username, IP, health check - whatever you think is necessary :p
